lib/spack/spack/spec.py

```diff
@@ -970,7 +970,115 @@ def __init__(self, spec, name, query_parameters):
 @key_ordering
 class Spec(object):
 
-    def __init__(self, spec_like, *dep_like, **kwargs):
+    @staticmethod
+    def from_literal(spec_dict, unique=True):
```

Comment:
What's the matter with `from_dict`?  `from_literal` seems kind of general -- there are many types of literals for specs, e.g. `libdwarf +foo ^libelf`.  Of course, there are also the JSON/YAML dicts that we use, so you could say that `from_dict` is ambiguous that way... hm.  Thoughts?  I still kind of like `from_dict`.  @adamjstewart: care to break a tie?